how-tos/providing-images-for-img.rst

```diff
@@ -0,0 +1,143 @@
+Providing images for <img>
+==========================
+
+:author: `pkel <https://github.com/pkel>`_
+
+In this tutorial, you will learn how to create an endpoint for providing images to HTML :code:<img> tags without client side javascript.
+The resulting HTML might look like this:
+
+.. code-block:: html
+
+   <img src="http://host/files/42/cats.jpeg" alt="Cute Kittens"/>
+
+In fact, the presented technique is suitable for providing not only images, but arbitrary files.
+
+We will start with a minimal example that highlights the general concept.
+Afterwards we present are more detailed solution that fixes a few shortcomings of the first approach.
+
+Minimal Example
+---------------
+
+PostgREST returns binary data on requests that set the :code:`Accept: application/octet-stream` header.
+The general idea is to configure the reverse proxy in front of the API to set this header for all requests to :code:`/files/`.
+We will show how to achieve this using Nginx.
+
+First, we need a public table for storing the files.
+
+.. code-block:: postgres
+   create table public.files(
+     id   int primary key
+   , blob bytea
+   );
+
+Let's assume this table contains an image of two cute kittens with id 42.
+We can retrieve this image in binary format from our PostgREST API by requesting :code:`/files?select=blog&id=eq.42` with the :code:`Accept: application/octet-stream` header.
+Unfortunately, putting the URL into the :code:`src` of an :code:`<img>` tag will not work.
+That's because browsers do not send the required header.
+
+Luckily, we can configure our Nginx reverse proxy to fix this problem for us.
+The following recipe assumes that PostgREST is running on port 3000.
+It configures Nginx as ordinary HTTP server on port 80.
+Requests to :code:`/api/*` are forwarded to our PostgREST instance.
+Requests to :code:`/files/<id>*` are forwarded to our endpoint with the :code:`Accept` header set to :code:`application/octet-stream`.
+
+.. code-block:: nginx
+   server {
+     listen 80 default_server;
+     listen [::]:80 default_server;
+     server_name _;
+     root /var/www/html;
+     index index.html;
+     try_files $uri $uri/ =404;
+
+     location /api/ {
+       default_type  application/json;
+       proxy_hide_header Content-Location;
+       add_header Content-Location /api/$upstream_http_content_location;
+       proxy_set_header  Connection "";
+       proxy_http_version 1.1;
+       proxy_pass http://localhost:3000/;
+     }
```

Comment:
I think all of this block is non-essential for the how-to, how about if you omit it?

Reply:
D'accord